src/viam/examples/modules/simple/main.cpp

```diff
@@ -24,118 +23,126 @@
                              const ::viam::common::v1::DoCommandRequest* request,
                              ::viam::common::v1::DoCommandResponse* response) override {
-        std::cout << "Received DoCommand request for MyModule number " << inner_which_
-                  << " and name " << name_ << std::endl;
+        std::cout << "Received DoCommand request for Printer " << name_ << std::endl;
         for (const auto& req : request->command().fields()) {
             std::cout << "request key: " << req.first.c_str()
                       << "\trequest value: " << req.second.SerializeAsString();
         }
         *response->mutable_result() = request->command();
 
+        std::cout << "Printer " << name_ << " has printed " << to_print_ << std::endl;
         return grpc::Status();
     }
 
+    static std::string find_to_print(ResourceConfig cfg) {
+        auto printer_name = cfg.name();
+        auto to_print = cfg.attributes()->find("to_print");
+        if (to_print == cfg.attributes()->end()) {
+            std::ostringstream buffer;
+            buffer << printer_name << ": Required parameter `to_print` not found in configuration";
+            throw std::invalid_argument(buffer.str());
+        }
+        const auto* const to_print_string = to_print->second->get<std::string>();
+        if (!to_print_string || to_print_string->empty()) {
+            std::ostringstream buffer;
+            buffer << printer_name
+                   << ": Required non-empty string parameter `to_print` is either not a string "
+                      "or is an empty string";
+            throw std::invalid_argument(buffer.str());
+        }
+        return *to_print_string;
+    }
+
    private:
     std::string name_;
-    static int which_;
-    int inner_which_;
+    std::string to_print_;
 };
 
-int MyModule::which_ = 0;
-
 int main(int argc, char** argv) {
     if (argc < 2) {
-        throw "need socket path as command line argument";
+        throw std::invalid_argument("Need socket path as command line argument");
     }
+    std::string socket_addr = argv[1];
 
     // Use set_logger_severity_from_args to set the boost trivial logger's
     // severity depending on commandline arguments.
     set_logger_severity_from_args(argc, argv);
     BOOST_LOG_TRIVIAL(debug) << "Starting module with debug level logging";
 
-    // C++ modules must handle SIGINT and SIGTERM. Make sure to create a sigset
-    // for SIGINT and SIGTERM that can be later awaited in a thread that cleanly
-    // shuts down your module. pthread_sigmask should be called near the start
-    // of main so that later threads inherit the mask.
-    sigset_t sigset;
-    sigemptyset(&sigset);
-    sigaddset(&sigset, SIGINT);
-    sigaddset(&sigset, SIGTERM);
-    pthread_sigmask(SIG_BLOCK, &sigset, NULL);
+    // C++ modules must handle SIGINT and SIGTERM. You can use the SignalManager
+    // class and its wait method to handle the correct signals.
+    SignalManager signals;
```

Comment:
I copied the signal handling logic from @abe-winter's wifi example CPP module. This should offset some of the burden of boilerplate signal handling from the user.